cmd/slirp-hook-sidecar/BUILD.bazel

```diff
@@ -0,0 +1,48 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_binary", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["slirp.go"],
+    importpath = "kubevirt.io/kubevirt/cmd/slirp-hook-sidecar",
+    visibility = ["//visibility:private"],
+    deps = [
+        "//cmd/slirp-hook-sidecar/dns:go_default_library",
+        "//cmd/slirp-hook-sidecar/server:go_default_library",
+        "//pkg/hooks:go_default_library",
+        "//pkg/hooks/info:go_default_library",
+        "//pkg/hooks/v1alpha2:go_default_library",
+        "//staging/src/kubevirt.io/client-go/log:go_default_library",
+        "@org_golang_google_grpc//:go_default_library",
+    ],
+)
+
+go_binary(
+    name = "slirp-hook-sidecar",
+    embed = [":go_default_library"],
+    visibility = ["//visibility:public"],
+)
+
+load(
+    "@io_bazel_rules_docker//container:container.bzl",
+    "container_image",
+)
+
+container_image(
+    name = "version-container",
+    base = "//:passwd-image",
+    directory = "/",
+    files = ["//:get-version"],
+)
+
+container_image(
+    name = "slirp-hook-sidecar-image",
+    architecture = select({
+        "@io_bazel_rules_go//go/platform:linux_arm64": "arm64",
+        "//conditions:default": "amd64",
+    }),
+    base = ":version-container",
```

Comment:
Hey, https://github.com/kubevirt/kubevirt/pull/9231 just got merged and that introduces a sidecar-shim base image that you could use and reduce some of the grpc boilerplate. [This commit](https://github.com/kubevirt/kubevirt/commit/d6950fb292b72f3aa9a291a264f76fadfef5fcb9) is the change for the smbios example

Reply:
Hi Victor, thanks for the heads up!

This slirp hook sidecar will be used as a reference for other network binding plugins (based on Kubevirt hook points and sidecar containers).
I am not sure we have the new sidecar-shim image available on potential D/S implementations, thus we decided to start with the "classic" hook implantation.

We will revisit the usage of sidecar-shim as a follow-up